pkg/build/generator/generator.go

```diff
@@ -261,6 +261,16 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	}
 	glog.V(4).Infof("Build %s/%s has been generated from %s/%s BuildConfig", newBuild.Namespace, newBuild.ObjectMeta.Name, bc.Namespace, bc.ObjectMeta.Name)
 
+	// Add the BuildConfig to the Build's OwnerReference, so that it will be
+	// deleted when the BuildConfig is deleted
+	// TODO: is there a utility function for generating an owner reference from an object?
+	newBuild.OwnerReferences = append(newBuild.OwnerReferences, kapi.OwnerReference{
+		APIVersion: "v1",          // BuildConfig.APIVersion is not populated
```

Comment:
> @deads2k that's got some interesting migration implications......

1. release 1: make new available, tolerate both
2. release 2: stop writing old, write only new, migrate old data
3. release 3: stop reading old and remove code cruft

We could stretch it out if we really wanted to I guess, but I think that's enough.